app/models/foreman_xen/xenserver.rb

```diff
@@ -60,12 +60,18 @@ def test_connection(options = {})
     end
 
     def available_hypervisors
+      Rails.cache.fetch(:available_hypervisors) { available_hypervisors! }
```

Comment:
I think, you should prefix the cache keys with the compute resource id.
It might be a good idea to define a method to do this for you

```ruby
def read_from_cache(key, fallback)
   Rails.cache.fetch(cache_key + key) { public_send(:fallback) }
end

def cache_key
  "computeresource_#{id}/"
end
```